feat/#2 Link 관련 기능 개발

Yu-Jaeyoung opened the pull request:
<!--아래 양식에 따라 작성하되 불필요한 항목은 삭제하고, 필요한 항목은 추가한다.-->

# 요구사항 및 기능 요약

Link 관련 기능 개발 진행
<br>

# 코드 리뷰 시 참고 사항
<!--리뷰어가 빠르게 코드 리뷰를 할 수 있도록 설명
예) abc.java의 L10-50은 코드 포맷 변경이므로 리뷰 필요 없음-->
FolderSerivce → folderRepository.save() 추가 필요 (추후 작업 예정)

Yu-Jaeyoung committed:
feat: folder InitDB 추가

- Folder에 InitDB를 위한 temp constructor 추가
- FolderCreateResponse 필드 명칭 수정
  - username → nickname

Yu-Jaeyoung committed:
feat: Link 기능 추가

- createLink() 등 개발

Resolves: #2

jckim22 commented on wafo-server/src/main/java/wafo/wafoserver/link/service/LinkService.java:
```diff
@@ -0,0 +1,47 @@
+    public LinkCreateResponse create(LinkCreateRequest linkCreateRequest) {
+
+        User user = userRepository.findById(linkCreateRequest.getUserId()).get();
+        Folder folder = folderRepository.findById(linkCreateRequest.getFolderId()).get();
```
이 부분에 디폴트 폴더가 들어가는 거라면

LinkCreateRequest에 상수값을 넣어주거나 
아래처럼 상수값을 사용해도 좋을 것 같습니다 

그럴려면 InitDB에 디폴트 폴더를 만들어줘야할 거 같습니다.

```suggestion
Folder folder = folderRepository.findById(DEFAULT_FOLDER_ID).get();
```

Yu-Jaeyoung replied:
현재 InitDB에 디폴트 폴더는 만들어 놓았습니다. *이전 커밋 참조

현재 로직은 경우의 수가 두 가지라고 생각합니다.
1. 사용자가 폴더를 미지정한 경우
2. 사용자가 폴더를 지정한 경우

이에 대한 공통적인 처리를 위해 위와 같이 작성하였습니다.

다만, 현재는 무조건 디폴트 폴더에 저장하는 형식으로 작성 후
추후에 LinkCreateRequest 등에서
사용자가 저장하려는 폴더를 지정 / 미지정했는지 판단한 후
올바른 폴더에 Link가 저장될 수 있도록 기능 수정하겠습니다.

jckim22 commented on wafo-server/src/main/java/wafo/wafoserver/link/service/LinkService.java:
```diff
@@ -0,0 +1,47 @@
+                        linkCreateRequest.getFaviconUrl()
+                )
+        );
+
```
링크를 트랜잭션 안에서 persist 하는 로직도 추가되어야 할 것 같습니다 !

```suggestion
linkrepository.save(link);
```

아래 테스트가 성공한 이유는 영속화 되지 않은 link를 그대로 DTO로 반환했기 때문에 값은 잘 나와서 통과한 것 같습니다.

Yu-Jaeyoung replied:
반영하였습니다 😁

jckim22 reviewed (commented):
리뷰 달았습니다.
나머지 부분이랑 API는 제가 완성할게욥

Yu-Jaeyoung committed:
fix: DEFAULT_FOLDER_ID 상수 지정

- defaultFolder에 Link 저장을 위해 ID 값을 상수로 지정
- LinkCreateResponse 필드 수정

Related to: #2

Yu-Jaeyoung commented:
리뷰 사항 반영하였습니다.
